arches/app/utils/i18n.py

```diff
@@ -10,9 +10,44 @@
 from arches.app.models.models import CardModel, CardXNodeXWidget, GraphModel, Language, PublishedGraph
 from django.utils.translation import get_language, get_language_info
 
+from arches.app.utils.betterJSONSerializer import JSONDeserializer, JSONSerializer
+
 ArchesPOFile = namedtuple("ArchesPOFile", ["language", "file"])
 
 
+def localize_serialized_object(json):
+    """
+    This method accepts a a serialized object or objects ( eg. the output of JSONSerializer().serialize() )
+    and recursively updates the serialized objects' key/value pairs to return only the localized
+    version of any i18n data
+
+    Arguments:
+    json -- (required) A JSON string representing a serialized object or objects
+
+    Returns:
+    A JSON string representing the input json object/objects, with any internationalized string/text dictionaries
+    localized to a string in the requested language, with the application language as a fallback.
+    """
+    deserialized_json = JSONDeserializer().deserialize(json)
+
+    def recursive_localize(deserialized_json):
+        if isinstance(deserialized_json, list):
+            for obj in deserialized_json:
+                recursive_localize(obj)
+        if isinstance(deserialized_json, dict):
+            for key in deserialized_json.keys():
+                try:
```

Comment:
This calls `get_localized_value` on every `dict`, and if `get_localized_value` throws an exception then the `dict` wasn't an i18n dict. It's possible to add some quick regex here to look at each key in the `dict` and if any of them match the i18n key pattern ('xx", 'xx-XX') only then call `get_localized_value`, but I'm unsure if it's worth it since `get_localized_value` is speedy enough on its own and it already iterates all the keys of the dict.